File: src/main/java/org/noxet/noxetserver/messaging/channels/MessagingPlayersChannel.java
package org.noxet.noxetserver.messaging.channels;

import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;

import java.util.HashSet;
import java.util.Set;

public class MessagingPlayersChannel extends MessagingChannel {
    private final Set<Player> players;

    public MessagingPlayersChannel(Set<Player> players) {
        this.players = players;
    }

    @Override
    public Set<CommandSender> getRecipients() {
        return new HashSet<>(players);
    }
}

File: src/main/java/org/noxet/noxetserver/messaging/channels/MessagingPlayersChannel.scala
package org.noxet.noxetserver.messaging.channels

import org.bukkit.command.CommandSender
import org.bukkit.entity.Player

import scala.collection.Set

class MessagingPlayersChannel(players: Set[Player]) extends MessagingChannel:
    override def getRecipients: Set[CommandSender] = players